src/main/kotlin/com/greenbird/utilihive/stubs/sftp/SftpServerTestContext.kt

```diff
@@ -0,0 +1,276 @@
+        encoding: Charset = UTF_8
+    ) {
+        val contentAsBytes = content.toByteArray(encoding)
+        putFile(path, ByteArrayInputStream(contentAsBytes))
+    }
+
+    /**
+     * Puts a file to the SFTP folder. The file is available by the specified
+     * path. The file's content is read from an `InputStream`.
+     * @param path the path to the file
+     * @param inputStream an `InputStream` that provides the file's content
+     */
+    fun putFile(
+        path: String,
+        inputStream: InputStream
+    ) {
+        verifyWithSftpServerIsNotFinished("upload file")
+        val pathAsObject = fileSystem.getPath(path)
+        ensureDirectoryOfPathExists(pathAsObject)
+        Files.copy(inputStream, pathAsObject)
+    }
+
+    /**
+     * Creates a directory on the SFTP server.
+     * @param path the directory's path
+     */
+    fun createDirectory(
+        path: String
+    ) {
+        verifyWithSftpServerIsNotFinished("create directory")
+        val pathAsObject = fileSystem.getPath(path)
+        Files.createDirectories(pathAsObject)
+    }
+
+    /**
+     * Create multiple directories on the SFTP server.
+     * @param paths the directories' paths.
+     */
+    fun createDirectories(
+        vararg paths: String
+    ) {
+        for (path in paths)
+            createDirectory(path)
+    }
+
+    /**
+     * Gets a text file's content from the SFTP server. The content is decoded
+     * using the specified encoding (UTF-8 if not specified).
+     * @param path the path to the file
+     * @param encoding the file's encoding
+     * @return the content of the text file
+     */
+    fun getFileText(
+        path: String,
+        encoding: Charset = UTF_8
+    ): String = getFileBytes(path).toString(encoding)
+
+    /**
+     * Gets a file from the SFTP server.
+     * @param path the path to the file
+     * @return the byte array with content of the file
+     */
+    fun getFileBytes(
+        path: String
+    ): ByteArray {
+        verifyWithSftpServerIsNotFinished("download file")
+        val pathAsObject = fileSystem.getPath(path)
+        return Files.readAllBytes(pathAsObject)
+    }
+
+    /**
+     * Converts a path string, or a sequence of strings that when joined form
+     * a path string, to a Path. If more does not specify any elements then
+     * the value of the first parameter is the path string to convert.
+     * If more specifies one or more elements, then each non-empty string,
+     * including first, is considered to be a sequence of name elements (see Path)
+     * and is joined to form a path string. Strings are joined using '/' as the separator.
+     * For example, if getPath("/foo","bar","gus") is invoked, then the path string
+     * "/foo/bar/gus" is converted to a Path. A Path representing an empty path is returned
+     * if first is the empty string and more does not contain any non-empty strings.
+     */
+    fun getPath(first: String, vararg more: String): Path {
```

Comment:
Here we use a lot of text explaining the arguments, but do not mention what this function is useful for. I.e. for creating a Path instance on the SFTP server you can use to list directories, inspect files etc. Add a sentence about that first?

Reply:
Done. I could also skip the second (vararg) argument if you want to simplify it, I only did it to make it more consistent with FileSystem.getPath method